packages/liveblocks-react/src/room.tsx

```diff
@@ -2357,14 +2359,19 @@ function makeRoomContextBundle<
 // ---------------------------------------------------------------------- }}}
 // --- Public APIs ------------------------------------------------------ {{{
 
+function useRoomOrNull() {
+  return React.useContext(RoomContext);
+}
+
 /**
  * @private
  *
  * This is an internal API, use `createRoomContext` instead.
  */
 export function useRoomContextBundleOrNull() {
   const client = useClientOrNull();
-  return client === null ? null : getOrCreateRoomContextBundle(client);
+  const room = useRoomOrNull();
```

Comment:
The key line for this fix. `useRoomContextBundleOrNull()` isn't really dependent on having a room in order to create the bundle, but… the way the `useSharedBundle()` function works is that it only expects this function to return a result if a RoomProvider context is active. This was a tricky one! Super glad you discovered this, @marcbouchenoire.

Next week, I'll ensure that the E2E page/test will also capture this edge case before I continue to refactor more.